Userland/Applications/PDFViewer/PDFViewer.cpp

```diff
@@ -247,3 +306,61 @@ PDF::PDFErrorOr<NonnullRefPtr<Gfx::Bitmap>> PDFViewer::render_page(const PDF::Pa
 
     return bitmap;
 }
+
+PDF::PDFErrorOr<void> PDFViewer::cache_page_dimensions(bool recalculate_fixed_info)
+{
+    if (recalculate_fixed_info)
+        m_page_dimension_cache.page_info.clear_with_capacity();
+
+    if (m_page_dimension_cache.page_info.is_empty()) {
+        m_page_dimension_cache.page_info.ensure_capacity(m_document->get_page_count());
+        for (size_t i = 0; i < m_document->get_page_count(); i++) {
```

Comment:
This forces us to read all page objects, of which there could be hundreds of thousands. This was identified as one of the problems in #23036. Could we instead compute information for pages lazily as needed, so we only have to read a few pages around the current page?

Reply:
Yeah I had started working on that, I can resurrect it if it looks like we do actually want to do that after all

Reply:
(To be clear, I meant do it lazily at the Applications/PDFViewer level. If that's what you mean too, then I think that'd be great to do!)

Reply:
Ahh, didn't really look at the code block you quoted closely enough. No that's not what I meant, I haven't done any work towards that